tools/enso4igv/src/main/java/org/enso/tools/enso4igv/enso/EnsoStructure.java

```diff
@@ -1,16 +1,53 @@
 package org.enso.tools.enso4igv.enso;
 
-import java.util.Collections;
+import java.util.ArrayList;
 import java.util.List;
+import javax.swing.text.BadLocationException;
 import javax.swing.text.Document;
+import org.enso.compiler.core.EnsoParser;
+import org.enso.compiler.core.IR;
+import org.enso.compiler.core.IR$Module$Scope$Definition$Data;
+import org.enso.compiler.core.IR$Module$Scope$Definition$SugaredType;
 import org.netbeans.api.editor.mimelookup.MimeRegistration;
 import org.netbeans.api.lsp.StructureElement;
 import org.netbeans.spi.lsp.StructureProvider;
+import scala.collection.Iterator;
 
 @MimeRegistration(mimeType = "application/x-enso", service = StructureProvider.class)
 public final class EnsoStructure implements StructureProvider {
   @Override
   public List<StructureElement> getStructure(Document dcmnt) {
-    return Collections.singletonList(StructureProvider.newBuilder("example", StructureElement.Kind.Module).build());
+    try {
+      var parser = new EnsoParser();
+      var text = dcmnt.getText(0, dcmnt.getLength());
+      var moduleIr = parser.compile(text);
+      var arr = new ArrayList();
+      var it = moduleIr.bindings().iterator();
+      collectStructure(arr, moduleIr.bindings().iterator());
+      return arr;
+    } catch (BadLocationException ex) {
+      throw new IllegalStateException(ex);
+    }
+  }
+
+  private static void collectStructure(ArrayList arr, Iterator<? extends IR> it) {
+    while (it.hasNext()) {
+      var b = it.next();
+      collectStructureItem(arr, b);
+    }
+  }
+  private static void collectStructureItem(ArrayList arr, IR b) {
```

Comment:
A shortcut is fetching `BindingsMap` from the `moduleIr` and asking for `definedEntities`. Also, I assume that in the future we also want to be able to get imported symbols, right?